pkg/util/podstore/podstore_test.go

```diff
@@ -0,0 +1,617 @@
+	}{
+		{
+			name: "object is not Pod",
+			podStore: &PodStore{
+				timestampMap: map[types.UID]*podLocalTimestamps{},
+				clock:        clock.NewFakeClock(refTime),
+				pods:         cache.NewIndexer(podKeyFunc, cache.Indexers{podIPIndex: podIPIndexFunc}),
+			},
+			obj:                     node,
+			expectedTimestampMapLen: 0,
+		},
+		{
+			name: "Valid case for Pending Pod",
+			obj:  pod1,
+			podStore: &PodStore{
+				timestampMap: map[types.UID]*podLocalTimestamps{},
+				clock:        clock.NewFakeClock(refTime),
+				pods:         cache.NewIndexer(podKeyFunc, cache.Indexers{podIPIndex: podIPIndexFunc}),
+			},
+			expectedTimestampMapLen: 1,
+			expectedCreationTime:    refTime,
+		},
+		{
+			name: "Valid case for Running Pod",
+			obj:  pod3,
+			podStore: &PodStore{
+				timestampMap: map[types.UID]*podLocalTimestamps{},
+				clock:        clock.NewFakeClock(refTime),
+				pods:         cache.NewIndexer(podKeyFunc, cache.Indexers{podIPIndex: podIPIndexFunc}),
+			},
+			expectedTimestampMapLen: 1,
+			expectedCreationTime:    refTime.Add(-5 * time.Minute),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equal(t, 0, len(tt.podStore.timestampMap))
+			tt.podStore.onPodCreate(tt.obj)
+			assert.Equal(t, tt.expectedTimestampMapLen, len(tt.podStore.timestampMap))
+			if tt.expectedTimestampMapLen > 0 {
+				assert.Equal(t, tt.expectedTimestampMapLen, len(tt.podStore.pods.List()))
+				assert.Equal(t, tt.expectedCreationTime, tt.podStore.timestampMap[tt.obj.(*v1.Pod).UID].CreationTimestamp)
+			}
+		})
+	}
+}
+
+func Test_onPodDelete(t *testing.T) {
+	k8sClient := fake.NewSimpleClientset()
+	podInformer := coreinformers.NewPodInformer(k8sClient, metav1.NamespaceAll, 0, cache.Indexers{})
+	tests := []struct {
+		name              string
+		obj               interface{}
+		timeWait          time.Duration
+		expectedLength    int
+		expectedTimestamp *podLocalTimestamps
+	}{
+		{
+			name:              "object is neither Pod or DeletedFinalStateUnknown",
+			obj:               node,
+			timeWait:          0,
+			expectedLength:    0,
+			expectedTimestamp: &podLocalTimestamps{CreationTimestamp: refTime},
+		},
+		{
+			name:              "Pod is in prevPod and podsToDelete",
+			obj:               pod1,
+			timeWait:          5 * time.Minute,
+			expectedLength:    1,
+			expectedTimestamp: &podLocalTimestamps{CreationTimestamp: refTime, DeletionTimestamp: &refTime},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClock := clock.NewFakeClock(refTime)
+			podStore := NewPodStoreWithClock(podInformer, fakeClock)
+			err := podStore.pods.Add(pod1)
+			assert.NoError(t, err)
+			podStore.timestampMap = map[types.UID]*podLocalTimestamps{"pod1": {CreationTimestamp: refTime}}
+
+			expectedDeleteTime := refTime.Add(tt.timeWait)
+			podStore.onPodDelete(tt.obj)
+			assert.Equal(t, tt.expectedTimestamp, podStore.timestampMap["pod1"])
+			if tt.expectedLength != 0 {
```

Comment:
when your 2 test cases are that different, I am of the opinion that it is better to use 2 separate test case implementations, rather than trying to unify them

```go
t.Run(...)
t.Run(...)
```

Reply:
Fixed. Thanks.